### DISCUSSION TITLE
Consider adding a human-friendly version of the `rememberedUserSessionDuration` value for "Keep me logged in"

### DISCUSSION THREAD

**Discussion by aaronbushnell** [craftcms/cms#11594] [Ideas] [answered]
I think it'd be useful to be explicit about how long you'll be remembered when using the "Keep me logged in" feature. I could see users thinking it's indefinite, but it's determined by the value of `rememberedUserSessionDuration` (default is 14 days).

When I've used this and have been logged out after the duration I briefly think something wasn't functioning properly. This would remove that ambiguity, IMO.

That might require splitting the "Forgot your password?" link to accommodate, but I think it still looks okay:

![Screen Shot 2022-07-13 at 1 06 05 PM](https://user-images.githubusercontent.com/315202/178791318-184c73f4-05b8-4796-ade8-9fe68065363d.png)

**Comment by brandonkelly** [comment, marked as answer]
Made this change for Craft 4.2! #11630

**Reply by aaronbushnell** [reply]
Thanks so much, @brandonkelly! 🎉

**Reply by brandonkelly** [reply]
Craft 4.2.0 is out now with that change ✨

**Comment by simonkuran** [comment]
This isn’t outputting the same value as what's in my config. For example. If I set `rememberedUserSessionDuration` to 'P6M' the login page says "Keep me signed in for 5 months, 4 weeks, 2 days, and 23 hours". I have to enter 'P184DT1H' to get a clean '6 months' on the login page.

**Reply by brandonkelly** [reply]
I’m not able to reproduce this, but am pretty sure I can see how it could happen. Just fixed for the next release (e30eee0d4ce698ec74998ca6db8170e137aff92d).

To test early, change your `craftcms/cms` requirement in composer.json to `dev-develop as 4.2.0.2`, and run `composer update`.

**Reply by simonkuran** [reply]
That fixed it, thanks! 'P6M' is now "6 months".

I'm guessing this is more of a related feature request, but it would be nice if it didn't jump to using a larger unit so quickly. For example 'P10D'  is "1 week and 3 days", but the most human readable output would be "10 days". Same with 'P180D' outputting "5 months, 3 weeks, and 6 days", instead of "180 days". Is it possible to make Craft respect formatting I give it via a ISO 8601 duration?

**Reply by brandonkelly** [reply]
Yep, just made that the case! (992e764506fa54885aa617f39a41a66d76cb64d5)

**Reply by simonkuran** [reply]
That works great, thanks!

**Reply by brandonkelly** [reply]
Craft 4.2.1 is out with that change.
